frontend/app/routes/-authenticated.js

```diff
@@ -7,4 +9,13 @@ export default class AuthenticatedRoute extends BaseRoute {
   beforeModel(transition) {
     this.session.requireAuthentication(transition, "account.login");
   }
+
+  @action error(error, transition) {
+    if (error instanceof UnauthorizedError) {
+      this.transitionTo("account.login");
+      return false;
+    }
+
+    return true;
+  }
```

Comment:
Why aren't you using the example approach in the adapter? If you do a patch in an action this won't do anything. It will only work for model hooks.

Reply:
Just invalidating the session lead to the same result as requireAuthentication is in `beforeModel` and the API error is in the `model` hook. This is the only way I found to redirect if there is an authentication error.

Reply:
I added `this.session.invalidate()` to the if-clause. This way the next page-request will be caught by the `sesssion.requireAuthentication`.

Reply:
If simple auth is correctly configured `this.session.invalidate()` should be enough, this triggers a redirect.. However my question still stands: why are you not using the application adapters handle response method like in [the example by ember simple auth](https://github.com/simplabs/ember-simple-auth/blob/74dd9cd2af45d791e0367d9671146eaaba2266f4/packages/ember-simple-auth/addon/mixins/data-adapter-mixin.js#L60-L94). This here you will only handle errors that happen in the model hook.

Reply:
I will have to check my configuration then.

Concerning the adapter. If I do the stuff from the example in my adapters `handleResponse` I get an empty error (without redirect). But if I get the redirect to work, that wouldn't be a problem.

Reply:
I added the handleResponse code from simple-auth but I don't see a way to make the redirect happen automatically. I also updated to the latest version of simple-auth (before was a beta) but that didn't help either.

Do you have an idea, what I could have configured incorrectly? The addon doesn't seem to have a classic configuration (anymore) as we have with simple-auth-oidc.

Any help would be appreciated!

Reply:
The `rootURL` config option has to be correct. If you look at the [code here](https://github.com/simplabs/ember-simple-auth/blob/master/packages/ember-simple-auth/addon/services/session.js#L204) it should redirect automatically after invalidation.

If the redirect works you shouldn't need the error handler in the authenticated route model hook.

Reply:
Are the handlers being registered, is `_setupHandlers` called?